### PR TITLE
Use a single lifetime for MIR construction

### PR DESCRIPTION
**Pull request by Zoxc**
Builds on https://github.com/rust-lang/rust/pull/57214

r? @eddyb

### CODE REVIEW

File: src/librustc_mir/build/expr/as_temp.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use crate::hair::*;
 use rustc::middle::region;
 use rustc::mir::*;
 
-impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
+impl<'a, 'tcx> Builder<'a, 'tcx> {
```
Comment (eddyb) [inline src/librustc_mir/build/expr/as_temp.rs]: You can replace headers like these with `impl Builder<'_, 'tcx>` if you want.

File: src/librustc_mir/build/mod.rs [hunk 2]
```diff
@@ -268,10 +196,11 @@ fn create_constructor_shim<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 ///////////////////////////////////////////////////////////////////////////
 // BuildMir -- walks a crate, looking for fn items and methods to build MIR from
 
-fn liberated_closure_env_ty<'a, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>,
-                                            closure_expr_id: hir::HirId,
-                                            body_id: hir::BodyId)
-                                            -> Ty<'tcx> {
+fn liberated_closure_env_ty<'a, 'tcx>(
+    tcx: TyCtxt<'a, 'tcx, 'tcx>,
```
Comment (eddyb) [inline src/librustc_mir/build/mod.rs]: This can be `TyCtxt<'_, 'tcx, 'tcx>` without the `<'a, 'tcx>` on the previous line.

**Review by eddyb — APPROVED** [review]
LGTM, modulo potential further lifetime cleanups. cc @pnkfelix @nikomatsakis @matthewjasper

**Comment by Zoxc** [comment]
@rust-timer build c0d6d4a069cb719fc05ca0a0aa4700f4e4a16600

File: src/librustc_mir/build/expr/as_temp.rs (hunk 1 above)
Comment (Zoxc) [reply src/librustc_mir/build/expr/as_temp.rs]: I feel like this is a job for rustfmt, not me =P

Comment (eddyb) [reply src/librustc_mir/build/expr/as_temp.rs]: Not `rustfmt` but maybe `rustfix`. Don't know if we have a lint for it yet though.

**Review by eddyb — APPROVED** [review]

**Pull request merged**